test/junit/scala/collection/ToConserveTest.scala

```diff
@@ -0,0 +1,90 @@
+package scala.collection
+
+import org.junit.Assert.{assertNotSame, assertSame}
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.collection.{immutable => i, mutable => m}
+import scala.language.implicitConversions
+import scala.{collection => c}
+
+@RunWith(classOf[JUnit4])
+class ToConserveTest {
+  // scala/bug#12188
+  implicit def toAnyRefFactory[A, CC[_] <: AnyRef](factory: c.IterableFactory[CC]): c.Factory[A, AnyRef] =
+    c.IterableFactory.toFactory(factory)
+  implicit def toFactory[K, V, CC[_, _] <: AnyRef](factory: MapFactory[CC]): Factory[(K, V), AnyRef] =
+    c.MapFactory.toFactory(factory)
+
+  @Test def toConserveList: Unit = {
+    val l: c.Iterable[Int] = (1 to 3).toList
+
+    assertSame(l, l.toList)
+    assertSame(l, l.toSeq)
+    assertSame(l, l.toIterable)
+
+    assertSame(l, l.to(List))
+
+    assertSame(l, l.to(c.Iterable))
+    assertSame(l, l.to(i.Iterable))
+
+    assertSame(l, l.to(c.Seq))
+    assertSame(l, l.to(i.Seq))
+
+    assertSame(l, l.to(c.LinearSeq))
+    assertSame(l, l.to(i.LinearSeq))
+
+    assertSame(l, l.to(List))
+  }
+
+  @Test def toConserveImmutableHashSet: Unit = {
+    val s: c.Iterable[Int] = (1 to 10).to(immutable.HashSet)
+    assertSame(s, s.toSet)
+    assertSame(s, s.toIterable)
+
+    assertSame(s, s.to(c.Iterable))
+    assertSame(s, s.to(i.Iterable))
+
+    assertSame(s, s.to(c.Set))
+    assertSame(s, s.to(i.Set))
+
+    assertSame(s, s.to(i.HashSet))
+  }
+
+  @Test def toConserveImmutableHashMap: Unit = {
+    val m: c.Iterable[(Int, Int)] = (1 to 10).map(x => (x, x)).to(immutable.HashMap): i.Map[Int, Int]
+
+    assertSame(m, m.toMap)
+    assertSame(m, m.toIterable)
+
+    assertSame(m, m.to(c.Iterable))
+    assertSame(m, m.to(i.Iterable))
+
+    assertSame(m, m.to(c.Map))
+    assertSame(m, m.to(i.Map))
+
+    assertSame(m, m.to(i.HashMap))
+  }
+
+  @Test def toRebuildMutable: Unit = {
+    val s: c.Iterable[Int] = (1 to 3).to(m.HashSet)
+    assertSame(s, s.toIterable) // slightly inconsistent...
```

Comment:
should `toIterable` be deprecated?

Reply:
@NthPortal hmm... can you expand on that thought?

Reply:
all the other `toX` methods copy the collection if it is mutable; the fact that `toIterable` doesn't is inconsistent and confusing. I could imagine it causing bugs if someone doesn't know (e.g. I didn't until now). It's also confusing that it doesn't do the same thing as `to(Iterable)`.

either that or we consider it a bug and just change it to do what `to(Iterable)` does

Reply:
views, also:

```
scala 2.13.6> List(1).view.toIterable
val res0: scala.collection.SeqView[Int] = SeqView(<not computed>)

scala 2.13.6> List(1).view.to(Iterable)
val res1: Iterable[Int] = List(1)
```

Reply:
it's unclear to me if it's just an implementation detail of [`IterableOps`](https://www.scala-lang.org/api/current/scala/collection/IterableOps.html#toIterable:Iterable[A]) that ended up public instead of protected, or if it's meant to be part of the public API. it should really be named `asIterable`.

Reply:
@NthPortal I agree.

Most uses of `toIterable` in the collections can easily be removed (pushed a commit, only two remain). It's possible that `toIterable` has a role when writing extension methods over collections that also need to work with Strings and Arrays (https://github.com/scala/scala/blob/v2.13.6/src/library/scala/collection/generic/IsSeq.scala#L69), but I think it should be trated internal to the collections.

https://github.com/scala/collection-strawman/issues/134#issuecomment-319872977 is the reason for it to be public, but deprecating it seems right.

Reply:
I'll submit the deprecation in a separate PR.

Reply:
thanks for doing the digging to find that previous discussion :)